miden-lib/src/transaction/mod.rs

```diff
@@ -161,6 +160,47 @@ impl TransactionKernel {
             .expect("Invalid stack input")
     }
 
+    /// Extends the advice inputs with account data and Merkle proofs.
+    ///
+    /// Where:
+    /// - account_header is the header of the account which data will be used for the extension.
+    /// - account_code is the code of the account which will be used for the extension.
+    /// - storage_header is the header of the storage which data will be used for the extension.
+    /// - merkle_path is the authentication path from the account root of the block header to the
+    ///   account. It may not be specified if the function is used for advice inputs extension with
+    ///   foreign account data.
```

Comment:
Question: I may be forgetting something, but don't foreign accounts also need to be authenticated?

Reply:
Also, I would say "It may be omitted" instead of "It may not be specified".

Reply:
> Question: I may be forgetting something, but don't foreign accounts also need to be authenticated?

I'm not sure, looks like this authentication path should be used somewhere in base, but I can't find where. @igamigo can you say where are we using this `merkle_path` during the transaction execution? Maybe we should use it in tests, and I just missed where. 

In my statement I meant that if we use this function in the FPI tests (which use the same approach of providing account data to the advice inputs) we don't need to specify the `merkle_path`, because it is never used there. 
But it looks strange to me to add a doc comment for the procedure input just for that case. I think it will be better to roll back and make this field not optional again, but when this function is used in tests just provide an empty vector as a merkle path. In that case we will be able to get rid of this sentence at all and simplify the function.

Reply:
But don't we authenticate foreign accounts in the kernel just as we do with native accounts? So, we should need the authentication path for them too.

Reply:
By the authentication in kernel do you mean the `authenticate_account_origin` procedure [here](https://github.com/0xPolygonMiden/miden-base/blob/7a628d5268aa5661084df495fa03f6e14f568ced/miden-lib/asm/kernels/transaction/api.masm#L61)?

Reply:
> Edit: probably you meant the `validate_current_foreign_account` from [here](https://github.com/0xPolygonMiden/miden-base/blob/7a628d5268aa5661084df495fa03f6e14f568ced/miden-lib/asm/kernels/transaction/lib/account.masm#L1054).

Yes, that's the one I meant. Specifically, the `mtree_verify` on line 1072 would fail if there is no corresponding authentication path in the advice provider.

Reply:
Now I get it. Previously I used `.with_merkle_store(mock_chain.accounts().into())` line to provide the foreign account hash (and path) to the advice provider, but I managed to provide it using this field.

Reply:
> I'm not sure, looks like this authentication path should be used somewhere in base, but I can't find where. @igamigo can you say where are we using this merkle_path during the transaction execution? Maybe we should use it in tests, and I just missed where.

Sorry I missed this before. As mentioned, I was counting on this being mandatory as well so I also did not quite get why you turned it into an `Option` before; I thought this validation was maybe not needed on the tests since I recalled the account tree was hardcoded into the inputs anyway.